api/src/main/java/org/apache/iceberg/UpdateSchema.java

```diff
@@ -384,4 +386,24 @@ default UpdateSchema updateColumn(String name, Type.PrimitiveType newType, Strin
    *                                  with other additions, renames, or updates.
    */
   UpdateSchema unionByNameWith(Schema newSchema);
+
+  /**
+   * Set the identifier fields given a set of field names.
+   * See {@link Schema#identifierFieldIds()} to learn more about Iceberg identifier.
+   *
+   * @param names names of the columns to set as identifier fields
+   * @return this for method chaining
+   */
+  UpdateSchema setIdentifierFields(Set<String> names);
```

Comment:
Why require this to be a set instead of accepting any `Collection` or `Iterable`? Would a user never need to pass a `List`?

Reply:
`Set` was used to explicitly tell users that identifier fields need to be unique, instead of giving people an illusion that the update operation can still succeed with repeated value.

Technically we can use `List`, `Iterable` or `Collection`. If we would like to make a broader use case for the API, I can document the behavior in javadoc.

Reply:
I would prefer to use `Collection`. It's fine to collapse that into a set internally since a set of columns or a multi-set of columns has the same identity behavior. `Collection` is easier to use for the caller.